app/components/post_profile_picture/post_profile_picture.js

```diff
@@ -55,6 +55,7 @@ export default class PostProfilePicture extends PureComponent {
 
         if (fromWebHook && enablePostIconOverride) {
             const icon = overrideIconUrl ? {uri: overrideIconUrl} : webhookIcon;
+            const borderRadius = icon === webhookIcon ? ViewTypes.PROFILE_PICTURE_SIZE / 2 : 0;
```

Comment:
why the change in the borderRadius?

Reply:
When an emoji is shown instead of the common profile pictures, the border makes it a bit weird. After some discussion, we dropped the border for any override, emoji or not.

Reply:
could you please share a screenshot?

Reply:
![image](https://user-images.githubusercontent.com/51929404/60925599-aeea6580-a293-11e9-82ea-70b1a1dba2bf.png)

(from the webapp)

Reply:
Is that being done for all overridden profile pictures or just emojis though? I'd expect it to only change for emojis if that was the case

Reply:
All overriden profile pictures.

If not done this way, we'd either have to
 - heuristically determine if it's about an emoji by checking if the URL is relative (quick, dirty way) or
 -  send the information along to the apps (it's not sent anymore - the emoji key is removed from the message props on the server before dispatching). Considering that, IIRC during some conversation we thought of changing all overriden profile pictures along the way.

If we don't do that, I'm 2/5 on changing this back and using the heuristic way. Not terribly clean, but not bad. Importantly, we could change this in the future because the emoji info is stored in the message in the database.

@aaronrothschild

Reply:
Please run this by UX

Reply:
@asaadmahmood I can't find Andrew Brown's username here on GH.  Adding you as a reviewer.

Quick Summary:  The purpose of this change is to provide Some better slack compatibility for developers when using our API.  This change allows a developer to override the user's profile photo and use an Emoji as the profile photo for the incoming webhook payload.  Since an emoji is often square-shaped, we made a 2/5 decision to remove the "circular mask" that frames profile photos by default (see screenshot from krjn above) when an override emoji OR photo URL is used.  

Question:   When an image is provided as the override in an incoming hook, should we "force" the 'circular mask' onto the image so it looks like a 'regular' profile picture?

IMO - I think that, Yes, we should enforce the circular mask for overridden pictures - this adds some consistency.   

@krjn I'm impartial as to *how* it is done, but obviously prefer a way that lets us change this behaviour in the future.

Reply:
Prefer to have the circular mask, however, I'd reduce the size of the emojis so most emojis don't normally hit it. Other than that, looks good.

Reply:
Prefer to have the circular mask, however, I'd reduce the size of the emojis so most emojis don't normally hit it. Other than that, looks good.

Reply:
New screenshot and commit. The last message shows a case that shouldn't normally happen, but it's there to illustrate how the border is being applied for emojis. The second message is of an url-overriden profile picture.

![image](https://user-images.githubusercontent.com/51929404/61072108-2211ea00-a402-11e9-8a43-deb0e4aaa891.png)

Reply:
I 0/5 that we should use circular mask. My hesitation is emojis with non-transparent backgrounds or image emojis like the last item above will have the corners clipped rather than respecting the circular mask.

To avoid the odd clipping issue, I would recommend not using the circular mask and set the dimensions (as a square) smaller (28x28) than the user avatar.

**Pro** - we avoid any odd clipping of the emojis, I'm 4/5 that I want to see the emojis as they are and not run a chance of clipping them and they loose context since we don't control what users will upload. This is less of a risk with avatars since users can crop them in the UI to their liking.
**Pro** - I'm okay with these being slightly smaller since they are non-standard avatars and have the `bot` label present.
**Con** - 0/5 if the subtle difference in avatar sizes will seem like a bug rather than a conscious decision.

**Stubbed out some options based on the feedback** - my vote is #2 (28px - no mask)
![image](https://user-images.githubusercontent.com/48559658/61089670-e1b66a00-a409-11e9-9876-49980563accd.png)

Reply:
I changed it to the second option. I liked the result and after some thinking, I believe the lesser size for emojis won't look weird/buggy.

Reply:
@andrewbrown00 @asaadmahmood @aaronrothschild 

1/5 I'm wondering why we need to special case it for emojis..if I have a trophy as my profile pic, it would cut off as well because of the circular border, I think? Or is the inconsistency not really a concern in this case?

Reply:
> 1/5 I'm wondering why we need to special case it for emojis..if I have a trophy as my profile pic, it would cut off as well because of the circular border, I think? Or is the inconsistency not really a concern in this case?

@jasonblais In the case of a profile pic - you are able to choose and see the result of uploading a photo and then see how the circle gets applied, and then you could adjust the size of the trophy for example and reupload.   

In this case, we are just using an Emoji programmatically - and there is no UI to preview what it will look like.  A developer could not reasonably expect that an emoji would have a somewhat arbitrary circle applied around it IMO.

In Slack, whose functionality we are trying to replicate here on this ticket - would not apply any kind of mask to a post that uses an Emoji, so let's emulate that functionality.

Reply:
> [...] so let's emulate that functionality.

After removing the mask, do we keep the 4px difference? (emojis being 28px instead of 32px)

Reply:
@aaronrothschild Thank you! Good points raised here

Reply:
> > [...] so let's emulate that functionality.
> 
> After removing the mask, do we keep the 4px difference? (emojis being 28px instead of 32px)

I'm going to defer to @andrewbrown00 earlier:
>To avoid the odd clipping issue, I would recommend not using the circular mask and set the dimensions (as a square) smaller (28x28) than the user avatar.  

Does that answer your question @krjn?

Reply:
Yep. It is ready then.